daemon/monitor.go

```diff
@@ -25,16 +25,19 @@ func (daemon *Daemon) setStateCounter(c *container.Container) {
 }
 
 func (daemon *Daemon) handleContainerExit(c *container.Container, e *libcontainerdtypes.EventInfo) error {
-	c.Lock()
-
-	ec, et, err := daemon.containerd.DeleteTask(context.Background(), c.ID)
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	ec, et, err := daemon.containerd.DeleteTask(ctx, c.ID)
+	cancel()
 	if err != nil {
-		logrus.WithError(err).Warnf("failed to delete container %s from containerd", c.ID)
+		logrus.WithError(err).WithField("container", c.ID).Warnf("failed to delete container from containerd")
+		return err
```

Comment:
I believe we should also be returning here if the function fails, as the rest of handleContainerExit is irrelevant if DeleteTask fails afaict

Reply:
The problem is we'll only run this once because we only get one event.
Perhaps now that (on master) we call `handleContainerExit` for `docker stop` regardless it could be ok... but it's better to clean up the reset of the resources.

Reply:
okay, I will remove the return then